srcts/src/components/darkModeSwitch.ts

```diff
@@ -0,0 +1,298 @@
+import type { CSSResultGroup, LitElement } from "lit";
+import { css, html } from "lit";
+import { property } from "lit/decorators.js";
+import { BslibElement } from "./webcomponents/_bslibElement";
+
+// Inspired by:
+// https://web.dev/building-a-theme-switch-component/
+// https://web.dev/patterns/theming/theme-switch/
+// https://github.com/argyleink/gui-challenges/tree/main/theme-switch
+
+type ToggleMessage = {
+  method: "toggle";
+  value: DarkModeSwitch["themeValue"] | "toggle";
+};
+
+export class DarkModeSwitch extends BslibElement {
+  static isShinyInput = true;
+  static tagName = "bslib-dark-mode-switch";
+  private themeAttribute = "data-shinytheme";
+
+  @property({ type: String, attribute: "theme-value", reflect: true })
+  // eslint-disable-next-line indent
+  themeValue: "dark" | "light" = "light";
+
+  static styles: CSSResultGroup = [
+    // CSS Variables
+    css`
+      :host {
+        /* open-props.style via shinycomponent */
+        --text-1: var(--text-1-light, var(--gray-8, #343a40));
+        --text-2: var(--text-2-light, var(--gray-7, #495057));
+        --size-xxs: var(--size-1, 0.25rem);
+        --ease-in-out-1: cubic-bezier(0.1, 0, 0.9, 1);
+        --ease-in-out-2: cubic-bezier(0.3, 0, 0.7, 1);
+
+        /* shinycomponent */
+        --speed-fast: 0.15s;
+        --speed-normal: 0.3s;
+
+        /* Move down to adjust for being large than 1em */
```

Comment:
```suggestion
```

Reply:
This is a comment from the original, so I'd prefer to leave it in place.

Reply:
Oh right, yeah that makes sense